pkg/controllers/webui/resource.go

```diff
@@ -58,8 +58,7 @@ func syncPodSpec(wi *v1alpha1.WebUI, dp *appsv1.Deployment) {
 }
 
 func getRollingUpdateStrategy(wi *v1alpha1.WebUI) *appsv1.RollingUpdateDeployment {
-	if wi.Spec.UpdateStrategy.MaxSurge != nil &&
-		wi.Spec.UpdateStrategy.MaxUnavailable != nil {
+	if wi.Spec.UpdateStrategy != nil {
```

Comment:
`UpdateStrategy != nil` does not imply `UpdateStrategy.MaxUnavailable != nil`, we should double check this to avoid nil pointer dereference